core/src/cluster_info_vote_listener.rs

```diff
@@ -836,16 +700,12 @@ impl ClusterInfoVoteListener {
         let mut new_optimistic_confirmed_slots = vec![];
 
         // Process votes from gossip and ReplayStage
-        for (is_gossip, (vote_pubkey, vote, _)) in gossip_vote_txs
+        let gossip_votes = gossip_vote_txs
             .iter()
-            .filter_map(|gossip_tx| {
-                vote_transaction::parse_vote_transaction(gossip_tx)
-                    .filter(|(vote_pubkey, vote, _)| {
-                        Self::filter_gossip_votes(vote_tracker, vote_pubkey, vote, gossip_tx)
-                    })
-                    .map(|v| (true, v))
-            })
-            .chain(replayed_votes.into_iter().map(|v| (false, v)))
+            .filter_map(vote_transaction::parse_vote_transaction);
+        for (is_gossip, (vote_pubkey, vote, _)) in repeat(true)
+            .zip(gossip_votes)
+            .chain(repeat(false).zip(replayed_votes))
```

Comment:
This is a little tricky to read.  wdyt about building the iterator outside the for-loop and swapping the `zip()`s so the loop iterates over `((vote_pubkey, vote, _), is_gossip)` ?

Reply:
done.

Reply:
Much easier on the eyes!  Thanks!